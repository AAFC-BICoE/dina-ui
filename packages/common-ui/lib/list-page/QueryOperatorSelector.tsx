import Select from "react-select";
import { useState } from "react";
import { FieldItems } from "react-awesome-query-builder";

interface QueryOperatorSelectorProps {
  options?: FieldItems;

  /**
   * Pass the selected option to the Query Builder.
   */
  setField: ((fieldPath: string) => void) | undefined;
}

interface QueryOperationOption {
  label: string;
  value: string;
}

export function QueryOperatorSelector({
  options,
  setField
}: QueryOperatorSelectorProps) {
  const [selectedOperator, setSelectedOperator] = useState<
    QueryOperationOption | undefined
  >();

  const customStyles = {
    placeholder: (provided, _) => ({
      ...provided,
      color: "rgb(87,120,94)"
    }),
    menu: (base) => ({ ...base, zIndex: 1050 }),
    control: (base) => ({
      ...base
    })
  };

  const operationOptions = options?.map<QueryOperationOption>((option) => ({
    label: option.label,
    value: option.path ?? ""
  }));

  return (
    <div style={{ width: "100%" }}>
      <Select<QueryOperationOption>
        options={operationOptions}
        className={`flex-grow-1 me-2 ps-0`}
        styles={customStyles}
        value={selectedOperator}
        onChange={(newValue) => {
          setSelectedOperator(newValue as QueryOperationOption);
          setField?.(newValue?.value ?? "");
        }}
      />
    </div>
  );
}
