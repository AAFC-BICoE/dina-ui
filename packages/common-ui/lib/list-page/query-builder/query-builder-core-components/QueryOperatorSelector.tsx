import Select from "react-select";
import { FieldItems } from "react-awesome-query-builder";
import { ESIndexMapping } from "../../types";

interface QueryOperatorSelectorProps {
  options?: FieldItems;

  /**
   * Operator selected from Query Builder.
   */
  selectedOperator?: string;

  /**
   * Pass the selected option to the Query Builder.
   */
  setOperator: ((fieldPath: string) => void) | undefined;

  /**
   * Field mapping for the selected field. This can determined what options are displayed
   * to the user for the operations.
   */
  selectedFieldMapping?: ESIndexMapping;
}

interface QueryOperationOption {
  label: string;
  value: string;
}

export function QueryOperatorSelector({
  options,
  selectedOperator,
  setOperator,
  selectedFieldMapping
}: QueryOperatorSelectorProps) {
  // Do not render if no operators are available, specifically the managed attributes.
  if (options?.length === 1 && options[0].key === "noOperator") {
    setOperator?.("noOperator");
    return <></>;
  }

  /* istanbul ignore next */
  const customStyles = {
    placeholder: (provided, _) => ({
      ...provided,
      color: "rgb(87,120,94)"
    }),
    menu: (base) => ({ ...base, zIndex: 1050 }),
    control: (base) => ({
      ...base
    })
  };

  // Some options are displayed only if it is supported.
  const operationOptions = options
    ?.filter((option) => {
      if (
        option.key === "startsWith" &&
        !selectedFieldMapping?.startsWithSupport
      ) {
        return false;
      }
      if (
        option.key === "containsText" &&
        !selectedFieldMapping?.containsSupport
      ) {
        return false;
      }
      if (option.key === "endsWith" && !selectedFieldMapping?.endsWithSupport) {
        return false;
      }
      return true;
    })
    ?.map<QueryOperationOption>((option) => ({
      label: option.label,
      value: option.key ?? ""
    }));

  const selectedOption = operationOptions?.find(
    (dropdownItem) => dropdownItem.value === selectedOperator
  );

  return (
    <div style={{ width: "100%" }}>
      <Select<QueryOperationOption>
        options={operationOptions}
        className={`flex-grow-1 me-2 ps-0`}
        styles={customStyles}
        value={selectedOption}
        onChange={(newValue) => setOperator?.(newValue?.value ?? "")}
      />
    </div>
  );
}
