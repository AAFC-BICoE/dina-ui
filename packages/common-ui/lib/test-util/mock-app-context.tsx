import { mount } from "enzyme";
import { merge, noop } from "lodash";
import { useMemo, useRef } from "react";
import { SWRConfig } from "swr";
import { PartialDeep } from "type-fest";
import { AccountContextI, AccountProvider } from "../account/AccountProvider";
import {
  ApiClientI,
  ApiClientImpl,
  ApiClientProvider
} from "../api-client/ApiClientContext";
import { CommonUIIntlProvider } from "../intl/common-ui-intl";
import { ModalProvider } from "../modal/modal";

interface MockAppContextProviderProps {
  apiContext?: PartialDeep<ApiClientI>;
  accountContext?: Partial<AccountContextI>;
  children?: React.ReactNode;
}

/**
 * Wraps a test-rendered component to provide the contexts that would be available in
 * the application.
 */
export function MockAppContextProvider({
  accountContext,
  apiContext = { apiClient: { get: () => undefined as any } },
  children
}: MockAppContextProviderProps) {
  const DEFAULT_MOCK_ACCOUNT_CONTEXT: AccountContextI = useMemo(
    () => ({
      authenticated: true,
      groupNames: ["aafc", "cnc"],
      initialized: true,
      login: noop,
      logout: noop,
      roles: ["user"],
      token: "test-token",
      username: "test-user",
      isAdmin: false
    }),
    []
  );

  const DEFAULT_API_CONTEXT_VALUE = useMemo(
    () =>
      new ApiClientImpl({
        newId: () => "00000000-0000-0000-0000-000000000000"
      }),
    []
  );

  const apiContextWithWarnings = {
    ...apiContext,
    // Add a warning when bulkGet doesn't return anything in a test:
    bulkGet: async (paths: string[], options) => {
      const resources = await apiContext?.bulkGet?.(paths, options);
      if (!resources) {
        console.warn("No response returned for bulkGet paths: ", paths);
      } else {
        resources.forEach((resource, index) => {
          if (!resource) {
            console.warn("No value returned for bulkGet path: ", paths[index]);
          }
        });
      }
      return resources;
    }
  };

  const modalWrapperRef = useRef<HTMLDivElement>(null);

  return (
    <SWRConfig
      // Reset SWR cache between tests.
      value={{ provider: () => new Map() }}
    >
      <AccountProvider
        value={{ ...DEFAULT_MOCK_ACCOUNT_CONTEXT, ...accountContext }}
      >
        <ApiClientProvider
          value={merge({}, DEFAULT_API_CONTEXT_VALUE, apiContextWithWarnings)}
        >
          <CommonUIIntlProvider>
            <div ref={modalWrapperRef}>
              <ModalProvider appElement={modalWrapperRef.current}>
                {children}
              </ModalProvider>
            </div>
          </CommonUIIntlProvider>
        </ApiClientProvider>
      </AccountProvider>
    </SWRConfig>
  );
}

/**
 * Helper function to get a test wrapper with the required context providers.
 */
export function mountWithAppContext(
  element: React.ReactNode,
  mockAppContextProviderProps?: MockAppContextProviderProps
) {
  return mount(
    <MockAppContextProvider {...mockAppContextProviderProps}>
      {element}
    </MockAppContextProvider>
  );
}
