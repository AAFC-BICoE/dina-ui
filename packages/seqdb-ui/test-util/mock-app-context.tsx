import {
  ApiClientContext,
  ApiClientContextI,
  createContextValue
} from "common-ui";
import { mount } from "enzyme";
import { SeqdbIntlProvider } from "../intl/seqdb-intl";

interface MockAppContextProviderProps {
  apiContext?: ApiClientContextI;
  children?: React.ReactNode;
}

/**
 * Wraps a test-rendered component to provide the contexts that would be available in
 * the application.
 */
export function MockAppContextProvider({
  apiContext,
  children
}: MockAppContextProviderProps) {
  return (
    <ApiClientContext.Provider value={apiContext || createContextValue()}>
      <SeqdbIntlProvider>{children}</SeqdbIntlProvider>
    </ApiClientContext.Provider>
  );
}

/**
 * Helper function to get a test wrapper with the required context providers.
 */
export function mountWithAppContext(
  element: React.ReactNode,
  mockAppContextProviderProps?: MockAppContextProviderProps
) {
  return mount(
    <MockAppContextProvider {...mockAppContextProviderProps}>
      {element}
    </MockAppContextProvider>
  );
}
