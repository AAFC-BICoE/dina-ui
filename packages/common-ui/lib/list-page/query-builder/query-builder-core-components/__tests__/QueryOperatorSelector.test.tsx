import { mountWithAppContext } from "common-ui/lib/test-util/mock-app-context";
import { useState } from "react";
import { FieldItems } from "react-awesome-query-builder";
import { QueryOperatorSelector } from "../QueryOperatorSelector";

const OPERATOR_OPTIONS: FieldItems = [
  {
    key: "exactMatch",
    label: "Exact Match"
  },
  {
    key: "partialMatch",
    label: "Partial Match"
  },
  {
    key: "notEquals",
    label: "Not Equals"
  },
  {
    key: "empty",
    label: "Empty"
  },
  {
    key: "notEmpty",
    label: "Not Empty"
  }
];

describe("QueryOperatorSelector component", () => {
  test("Snapshot Test", async () => {
    const wrapper = mountWithAppContext(
      <QueryOperatorSelector
        options={OPERATOR_OPTIONS}
        selectedOperator={OPERATOR_OPTIONS[0].key}
        setOperator={undefined}
      />
    );

    // Simulate opening up the menu.
    wrapper.find("DropdownIndicator").simulate("mouseDown", {
      button: 0
    });

    // Snapshot with all of the options and layout.
    expect(wrapper.find(QueryOperatorSelector).debug()).toMatchSnapshot();
  });

  test("Toggle between the options", async () => {
    function TestSelector() {
      const [operator, setOperator] = useState<string>(OPERATOR_OPTIONS[0].key);

      return (
        <QueryOperatorSelector
          options={OPERATOR_OPTIONS}
          selectedOperator={operator}
          setOperator={(newOperator) => setOperator(newOperator)}
        />
      );
    }

    const wrapper = mountWithAppContext(
      <>
        <TestSelector />
      </>
    );

    // Expect the initial state.
    expect(
      wrapper.find(QueryOperatorSelector).prop("selectedOperator")
    ).toEqual(OPERATOR_OPTIONS[0].key);

    // Select a new option in the list.
    wrapper.find("DropdownIndicator").simulate("mouseDown", {
      button: 0
    });
    wrapper.find("Option").at(1).find("div").simulate("click");
    wrapper.update();
    expect(
      wrapper.find(QueryOperatorSelector).prop("selectedOperator")
    ).toEqual(OPERATOR_OPTIONS[1].key);

    // And, select another option in the list.
    wrapper.find("DropdownIndicator").simulate("mouseDown", {
      button: 0
    });
    wrapper.find("Option").at(2).find("div").simulate("click");
    wrapper.update();
    expect(
      wrapper.find(QueryOperatorSelector).prop("selectedOperator")
    ).toEqual(OPERATOR_OPTIONS[2].key);
  });
});
