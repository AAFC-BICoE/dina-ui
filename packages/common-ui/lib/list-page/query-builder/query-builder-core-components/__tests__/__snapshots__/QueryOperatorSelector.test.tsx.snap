// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryOperatorSelector component Snapshot Test 1`] = `
"<QueryOperatorSelector options={{...}} selectedOperator=\\"exactMatch\\" setOperator={[undefined]}>
  <div style={{...}}>
    <ForwardRef options={{...}} className=\\"flex-grow-1 me-2 ps-0\\" styles={{...}} value={{...}} onChange={[Function: onChange]}>
      <Select options={{...}} className=\\"flex-grow-1 me-2 ps-0\\" styles={{...}} inputValue=\\"\\" menuIsOpen={true} onChange={[Function (anonymous)]} onInputChange={[Function (anonymous)]} onMenuClose={[Function (anonymous)]} onMenuOpen={[Function (anonymous)]} value={{...}} aria-live=\\"polite\\" backspaceRemovesValue={true} blurInputOnSelect={true} captureMenuScroll={false} closeMenuOnSelect={true} closeMenuOnScroll={false} components={{...}} controlShouldRenderValue={true} escapeClearsValue={false} filterOption={[Function (anonymous)]} formatGroupLabel={[Function: formatGroupLabel]} getOptionLabel={[Function: getOptionLabel]} getOptionValue={[Function: getOptionValue]} isDisabled={false} isLoading={false} isMulti={false} isRtl={false} isSearchable={true} isOptionDisabled={[Function: isOptionDisabled]} loadingMessage={[Function: loadingMessage]} maxMenuHeight={300} minMenuHeight={140} menuPlacement=\\"bottom\\" menuPosition=\\"absolute\\" menuShouldBlockScroll={false} menuShouldScrollIntoView={true} noOptionsMessage={[Function: noOptionsMessage]} openMenuOnFocus={false} openMenuOnClick={true} pageSize={5} placeholder=\\"Select...\\" screenReaderStatus={[Function: screenReaderStatus]} tabIndex={0} tabSelectsValue={true}>
        <SelectContainer clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} className=\\"flex-grow-1 me-2 ps-0\\" innerProps={{...}} isDisabled={false} isFocused={false}>
          <EmotionCssPropInternal css={{...}} className=\\"flex-grow-1 me-2 ps-0\\" id={[undefined]} onKeyDown={[Function (anonymous)]} __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"SelectContainer\\">
            <div className=\\"flex-grow-1 me-2 ps-0 css-b62m3t-container\\" id={[undefined]} onKeyDown={[Function (anonymous)]}>
              <LiveRegion clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} id=\\"react-select-2-live-region\\" ariaSelection={{...}} focusedOption={{...}} focusedValue={{...}} isFocused={false} selectValue={{...}} focusableOptions={{...}}>
                <A11yText id=\\"react-select-2-live-region\\">
                  <EmotionCssPropInternal css={{...}} id=\\"react-select-2-live-region\\" __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"span\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"A11yText\\">
                    <span id=\\"react-select-2-live-region\\" className=\\"css-1f43avz-a11yText-A11yText\\" />
                  </EmotionCssPropInternal>
                </A11yText>
                <A11yText aria-live=\\"polite\\" aria-atomic=\\"false\\" aria-relevant=\\"additions text\\">
                  <EmotionCssPropInternal css={{...}} aria-live=\\"polite\\" aria-atomic=\\"false\\" aria-relevant=\\"additions text\\" __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"span\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"A11yText\\">
                    <span aria-live=\\"polite\\" aria-atomic=\\"false\\" aria-relevant=\\"additions text\\" className=\\"css-1f43avz-a11yText-A11yText\\" />
                  </EmotionCssPropInternal>
                </A11yText>
              </LiveRegion>
              <Control clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} innerRef={[Function (anonymous)]} innerProps={{...}} isDisabled={false} isFocused={false} menuIsOpen={true}>
                <EmotionCssPropInternal css={{...}} className=\\"\\" onMouseDown={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"Control\\">
                  <div className=\\" css-1s2u09g-control\\" onMouseDown={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]}>
                    <ValueContainer clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} isDisabled={false}>
                      <EmotionCssPropInternal css={{...}} className=\\"\\" __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"ValueContainer\\">
                        <div className=\\" css-319lph-ValueContainer\\">
                          <SingleValue clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} data={{...}} isDisabled={false}>
                            <EmotionCssPropInternal css={{...}} className=\\"\\" __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"SingleValue\\">
                              <div className=\\" css-qc6sy-singleValue\\">
                                Exact Match
                              </div>
                            </EmotionCssPropInternal>
                          </SingleValue>
                          <Input clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} autoCapitalize=\\"none\\" autoComplete=\\"off\\" autoCorrect=\\"off\\" id=\\"react-select-2-input\\" innerRef={[Function (anonymous)]} isDisabled={false} isHidden={false} onBlur={[Function (anonymous)]} onChange={[Function (anonymous)]} onFocus={[Function (anonymous)]} spellCheck=\\"false\\" tabIndex={0} form={[undefined]} type=\\"text\\" value=\\"\\" aria-autocomplete=\\"list\\" aria-expanded={true} aria-haspopup={true} aria-controls=\\"react-select-2-listbox\\" aria-owns=\\"react-select-2-listbox\\" aria-errormessage={[undefined]} aria-invalid={[undefined]} aria-label={[undefined]} aria-labelledby={[undefined]} role=\\"combobox\\">
                            <EmotionCssPropInternal className=\\"\\" css={{...}} data-value=\\"\\" __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"Input\\">
                              <div className=\\" css-6j8wv5-Input\\" data-value=\\"\\">
                                <input className=\\"\\" style={{...}} disabled={false} autoCapitalize=\\"none\\" autoComplete=\\"off\\" autoCorrect=\\"off\\" id=\\"react-select-2-input\\" onBlur={[Function (anonymous)]} onChange={[Function (anonymous)]} onFocus={[Function (anonymous)]} spellCheck=\\"false\\" tabIndex={0} form={[undefined]} type=\\"text\\" value=\\"\\" aria-autocomplete=\\"list\\" aria-expanded={true} aria-haspopup={true} aria-controls=\\"react-select-2-listbox\\" aria-owns=\\"react-select-2-listbox\\" aria-errormessage={[undefined]} aria-invalid={[undefined]} aria-label={[undefined]} aria-labelledby={[undefined]} role=\\"combobox\\" />
                              </div>
                            </EmotionCssPropInternal>
                          </Input>
                        </div>
                      </EmotionCssPropInternal>
                    </ValueContainer>
                    <IndicatorsContainer clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} isDisabled={false}>
                      <EmotionCssPropInternal css={{...}} className=\\"\\" __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"IndicatorsContainer\\">
                        <div className=\\" css-1hb7zxy-IndicatorsContainer\\">
                          <IndicatorSeparator clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} isDisabled={false} isFocused={false}>
                            <EmotionCssPropInternal css={{...}} className=\\"\\" __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"span\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"IndicatorSeparator\\">
                              <span className=\\" css-1okebmr-indicatorSeparator\\" />
                            </EmotionCssPropInternal>
                          </IndicatorSeparator>
                          <DropdownIndicator clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} innerProps={{...}} isDisabled={false} isFocused={false}>
                            <EmotionCssPropInternal css={{...}} className=\\"\\" onMouseDown={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} aria-hidden=\\"true\\" __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"DropdownIndicator\\">
                              <div className=\\" css-tlfecz-indicatorContainer\\" onMouseDown={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} aria-hidden=\\"true\\">
                                <DownChevron>
                                  <Svg size={20}>
                                    <EmotionCssPropInternal height={20} width={20} viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" css={{...}} __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"svg\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"Svg\\">
                                      <svg height={20} width={20} viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"css-tj5bde-Svg\\">
                                        <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\" />
                                      </svg>
                                    </EmotionCssPropInternal>
                                  </Svg>
                                </DownChevron>
                              </div>
                            </EmotionCssPropInternal>
                          </DropdownIndicator>
                        </div>
                      </EmotionCssPropInternal>
                    </IndicatorsContainer>
                  </div>
                </EmotionCssPropInternal>
              </Control>
              <MenuPlacer clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} minMenuHeight={140} maxMenuHeight={300} menuPlacement=\\"bottom\\" menuPosition=\\"absolute\\" menuShouldScrollIntoView={true}>
                <Menu clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} minMenuHeight={140} maxMenuHeight={300} menuPlacement=\\"bottom\\" menuPosition=\\"absolute\\" menuShouldScrollIntoView={true} innerRef={[Function (anonymous)]} innerProps={{...}} isLoading={false} placement=\\"bottom\\">
                  <EmotionCssPropInternal css={{...}} className=\\"\\" onMouseDown={[Function (anonymous)]} onMouseMove={[Function (anonymous)]} id=\\"react-select-2-listbox\\" __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"Menu\\">
                    <div className=\\" css-blvick-menu\\" onMouseDown={[Function (anonymous)]} onMouseMove={[Function (anonymous)]} id=\\"react-select-2-listbox\\">
                      <ScrollManager captureEnabled={false} onTopArrive={[undefined]} onBottomArrive={[undefined]} lockEnabled={false}>
                        <MenuList clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} innerRef={[Function: innerRef]} isLoading={false} maxHeight={300} focusedOption={{...}}>
                          <EmotionCssPropInternal css={{...}} className=\\"\\" __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"MenuList\\">
                            <div className=\\" css-4ljt47-MenuList\\">
                              <Option clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} innerProps={{...}} data={{...}} isDisabled={false} isSelected={true} label=\\"Exact Match\\" type=\\"option\\" value=\\"exactMatch\\" isFocused={true} innerRef={[Function (anonymous)]}>
                                <EmotionCssPropInternal css={{...}} className=\\"\\" aria-disabled={false} id=\\"react-select-2-option-0\\" onClick={[Function (anonymous)]} onMouseMove={[Function (anonymous)]} onMouseOver={[Function (anonymous)]} tabIndex={-1} __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"Option\\">
                                  <div className=\\" css-9gakcf-option\\" aria-disabled={false} id=\\"react-select-2-option-0\\" onClick={[Function (anonymous)]} onMouseMove={[Function (anonymous)]} onMouseOver={[Function (anonymous)]} tabIndex={-1}>
                                    Exact Match
                                  </div>
                                </EmotionCssPropInternal>
                              </Option>
                              <Option clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} innerProps={{...}} data={{...}} isDisabled={false} isSelected={false} label=\\"Partial Match\\" type=\\"option\\" value=\\"partialMatch\\" isFocused={false} innerRef={[undefined]}>
                                <EmotionCssPropInternal css={{...}} className=\\"\\" aria-disabled={false} id=\\"react-select-2-option-1\\" onClick={[Function (anonymous)]} onMouseMove={[Function (anonymous)]} onMouseOver={[Function (anonymous)]} tabIndex={-1} __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"Option\\">
                                  <div className=\\" css-yt9ioa-option\\" aria-disabled={false} id=\\"react-select-2-option-1\\" onClick={[Function (anonymous)]} onMouseMove={[Function (anonymous)]} onMouseOver={[Function (anonymous)]} tabIndex={-1}>
                                    Partial Match
                                  </div>
                                </EmotionCssPropInternal>
                              </Option>
                              <Option clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} innerProps={{...}} data={{...}} isDisabled={false} isSelected={false} label=\\"Not Equals\\" type=\\"option\\" value=\\"notEquals\\" isFocused={false} innerRef={[undefined]}>
                                <EmotionCssPropInternal css={{...}} className=\\"\\" aria-disabled={false} id=\\"react-select-2-option-2\\" onClick={[Function (anonymous)]} onMouseMove={[Function (anonymous)]} onMouseOver={[Function (anonymous)]} tabIndex={-1} __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"Option\\">
                                  <div className=\\" css-yt9ioa-option\\" aria-disabled={false} id=\\"react-select-2-option-2\\" onClick={[Function (anonymous)]} onMouseMove={[Function (anonymous)]} onMouseOver={[Function (anonymous)]} tabIndex={-1}>
                                    Not Equals
                                  </div>
                                </EmotionCssPropInternal>
                              </Option>
                              <Option clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} innerProps={{...}} data={{...}} isDisabled={false} isSelected={false} label=\\"Empty\\" type=\\"option\\" value=\\"empty\\" isFocused={false} innerRef={[undefined]}>
                                <EmotionCssPropInternal css={{...}} className=\\"\\" aria-disabled={false} id=\\"react-select-2-option-3\\" onClick={[Function (anonymous)]} onMouseMove={[Function (anonymous)]} onMouseOver={[Function (anonymous)]} tabIndex={-1} __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"Option\\">
                                  <div className=\\" css-yt9ioa-option\\" aria-disabled={false} id=\\"react-select-2-option-3\\" onClick={[Function (anonymous)]} onMouseMove={[Function (anonymous)]} onMouseOver={[Function (anonymous)]} tabIndex={-1}>
                                    Empty
                                  </div>
                                </EmotionCssPropInternal>
                              </Option>
                              <Option clearValue={[Function (anonymous)]} cx={[Function (anonymous)]} getStyles={[Function (anonymous)]} getValue={[Function (anonymous)]} hasValue={true} isMulti={false} isRtl={false} options={{...}} selectOption={[Function (anonymous)]} selectProps={{...}} setValue={[Function (anonymous)]} theme={{...}} innerProps={{...}} data={{...}} isDisabled={false} isSelected={false} label=\\"Not Empty\\" type=\\"option\\" value=\\"notEmpty\\" isFocused={false} innerRef={[undefined]}>
                                <EmotionCssPropInternal css={{...}} className=\\"\\" aria-disabled={false} id=\\"react-select-2-option-4\\" onClick={[Function (anonymous)]} onMouseMove={[Function (anonymous)]} onMouseOver={[Function (anonymous)]} tabIndex={-1} __EMOTION_TYPE_PLEASE_DO_NOT_USE__=\\"div\\" __EMOTION_LABEL_PLEASE_DO_NOT_USE__=\\"Option\\">
                                  <div className=\\" css-yt9ioa-option\\" aria-disabled={false} id=\\"react-select-2-option-4\\" onClick={[Function (anonymous)]} onMouseMove={[Function (anonymous)]} onMouseOver={[Function (anonymous)]} tabIndex={-1}>
                                    Not Empty
                                  </div>
                                </EmotionCssPropInternal>
                              </Option>
                            </div>
                          </EmotionCssPropInternal>
                        </MenuList>
                      </ScrollManager>
                    </div>
                  </EmotionCssPropInternal>
                </Menu>
              </MenuPlacer>
            </div>
          </EmotionCssPropInternal>
        </SelectContainer>
      </Select>
    </ForwardRef>
  </div>
</QueryOperatorSelector>"
`;
